rdrf/rdrf/forms/dsl/code_evaluator.py

```diff
@@ -0,0 +1,142 @@
+from datetime import datetime
+from itertools import chain
+import logging
+
+from ..dynamic.value_fetcher import DynamicValueFetcher
+from .constants import VISIBLE, HIDDEN
+from .parse_operations import BooleanOp, Condition, parse_dsl, transform_tree
+from .parse_utils import CDEHelper, SectionHelper
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConditionEvaluator:
+
+    INTEGER_TYPE = 'integer'
+    DATE_TYPE = 'date'
+
+    def __init__(self, operator, value, existing_value, data_type):
+        self.operator = operator
+        self.value = value
+        self.existing_value = existing_value
+        self.data_type = data_type
+
+    @staticmethod
+    def parse_date(value):
```

Comment:
It's only used here so for now I thinks it doesn't need to be moved.

Reply:
Well, 2 problems with the approach:

- we risk keep implementing these on different classes/in different models because you look if it exists in `utils.py` you don't find it then you implement it
- I don't see how it is the responsibility of the `ConditionEvaluator` class to parse dates and ints. Why are they defined on the class then?

Reply:
Updated.